pkgs/applications/networking/browsers/chromium/common.nix

```diff
@@ -149,6 +149,10 @@ let
 
       sed -i -e '/libpci_loader.*Load/s!"\(libpci\.so\)!"${pciutils}/lib/\1!' \
         gpu/config/gpu_info_collector_linux.cc
+    '' + optionalString enableNaCl ''
+      cp -drT ${upstream-info.toolchain} native_client/toolchain
+      chmod -R u+w native_client/toolchain
+      find native_client/toolchain -type f -a -perm +0100 -exec "${./patch_toolchain.sh}" {} \;
```

Comment:
We could try that. This would also remove the need for `stdenv_32bit` as you mentioned below. https://www.chromium.org/nativeclient/pnacl/building-pnacl-components-for-distribution-packagers contains some notes on how to build the toolchains. I tried this approach back in June when I first started on this, but couldn't get it to work. It does seem like they've made some improvements since then though, so it might be worth a shot. Sadly there's no separate tar-balls for the toolchains though, we need to run a script supplied with the chromium source